libs/emblem/src/components/hooks/use-market-info.ts

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from 'react';
+import { URL_BASE } from '../../config';
+
+export type MarketInfo = {
+  quoteLogo: string;
+  settlementLogo?: string;
+  baseLogo: string;
+};
+
+export type MarketInfoState = {
+  loading: boolean;
+  data?: MarketInfo | null;
+  error: Error | null | undefined;
+};
+
+export async function fetchMarketInfo(path: string) {
+  return (await fetch(path)).json();
+}
+
+export const useMarketInfo = (vegaChain: string, marketId: string) => {
+  const [state, setState] = useState<MarketInfoState>({
+    loading: true,
+    data: null,
+    error: null,
+  });
+
+  useEffect(() => {
+    let mounted = true;
+
+    const path = `${URL_BASE}/vega/${vegaChain}/market/${marketId}/index.json`;
+
+    fetchMarketInfo(path)
```

Comment:
It seems sensible to have this only fetched and cached once? I think we do something similar for the oracle proofs json. Perhaps a helper hook for fetching and caching the result in memory for these types of things would be good